src/string.cr

```diff
@@ -1743,6 +1743,21 @@ class String
     end
   end
 
+  # Relturns a string where all strings in the given hash are replaced.
+  # by the corresponding *hash* values.
+  #
+  # ```
+  # "hello".gsub({"e" => 'a', "l" => 'd'}) # "haddo"
+  # ```
+  def gsub(hash : Hash(String, _))
+    string = self
+    hash.each do |key, value|
+      string = string.gsub(key, value)
```

Comment:
Definitely not as slow as I would have thought:

```cr
require "benchmark"

string  = "Apples Oranges " * 100
hash    = { "p" => 'a', "l" => 's', "es" => 'c', "Oranges" => 'f' }

inverse = hash.invert
string_inverse = string.gsub(hash)

class String
  def gsub(hash : Hash(String, _))
    string = self
    hash.each do |key, value|
      string = string.gsub(key, value)
    end

    string
  end
end

Benchmark.ips do |x|
  x.report("Hash(String => Char)") do
    string.gsub(hash)
  end

  x.report("Hash(Char => String)") do
    string_inverse.gsub(inverse)
  end
end
```
```
$ ./test
Hash(String => Char)  37.59k (± 0.59%)  1.82× slower
Hash(Char => String)   68.4k (± 0.86%)       fastest
```
With single char replacements:
```
$ ./test
Hash(String => Char)  40.93k (± 0.45%)  1.45× slower
Hash(Char => String)  59.46k (± 2.19%)       fastest
```

I reduced the string length, and used `Benchmark.ips` because it has a much nicer output. it's a bit slower but not by as much as I would have thought.